src/js/apps/globals/nav_app.js

```diff
@@ -154,28 +161,45 @@ export default App.extend({
   },
   stateEvents: {
     'change:currentApp': 'onChangeCurrentApp',
+    'change:isMinimized': 'onChangeIsMinimized',
   },
   viewEvents: {
     'click:addPatient': 'onClickAddPatient',
+    'click:minimizeMenu': 'onClickMinimizeMenu',
   },
   childApps: {
     search: SearchApp,
   },
   selectNav(appName, event, eventArgs) {
     this.setState('currentApp', appName);
 
-    const navMatch = this.getNavMatch(appName, event, compact(eventArgs));
+    this.navMatch = this.getNavMatch(appName, event, compact(eventArgs));
 
-    if (navMatch) {
+    if (this.navMatch) {
       this.getView().removeSelected();
-      navMatch.trigger('selected');
+      this.navMatch.trigger('selected');
     }
   },
   onChangeCurrentApp(state, appName) {
     this.setSelectedAdminNavItem(appName);
 
     this.getView().removeSelected();
   },
+  onChangeIsMinimized() {
+    this.showMainNavDroplist();
+    this.showBottomNavView();
+    this.showNav();
+
+    if (this.navMatch) {
+      this.getView().removeSelected();
+      this.navMatch.trigger('selected');
+    }
+
+    const currentAppName = this.getState('currentApp');
+    this.setSelectedAdminNavItem(currentAppName);
```

Comment:
`this.navMatch.trigger('selected'` preserves the selected nav item (left blue border added) when switching between the minimized and expanded versions of the nav.

`this.setSelectedAdminNavItem(currentAppName)` preserves the selected state in the admin tools droplist menu.